react/components/ExtensionPointComponent.tsx

```diff
@@ -213,12 +219,51 @@ class ExtensionPointComponent extends PureComponent<
       delete props.__clearError
     }
 
-    return Component ? (
-      <Component {...props}>{children}</Component>
-    ) : (
-      children || <Loading />
+    return (
+      <TreePathContext.Provider value={{ treePath }}>
```

Comment:
Use memo here

```tsx
const treePathValue = useMemo(() => ({ treePath }), [treePath])

<TreePathContext.Provider value={treePathValue}>
```

Reply:
This is a class component 😬 
I can make a simple cache logic.

Reply:
You can use https://www.npmjs.com/package/memoize-one for that

Reply:
You can also create a memoized provider

```tsx
`const TreePathContextProvider = memo(({ treePath, children }) =>
  <TreePathContext.Provider value={{ treePath }}>
    {children}
  </TreePathContext.Provider>
)
```